src/compile.rs

```diff
@@ -329,6 +347,158 @@ pub fn build_func_map<'a>(
+                                range = cas_arg.get_range();
+                            }
+                        }
+                    }
+                    error = Some(add_or_create_compile_error(error,
+                        "Duplicate filesystem context.",
+                        &rule.file,
+                        range,
+                        &format!("Found multiple different filesystem type declarations for filesystem: {}", rule.fs_name)));
+                }
+                FSContextType::GenFSCon => {
+                    // genfscon gets more complicated.  We can have similar rules as long as the paths are different.
+                    // If we find a genfscon with the same path, they must have the same context and object type.
+                    if let Some(path) = &rule.path {
+                        // Look through the rules again
+                        for inner_rule in &v {
+                            // Only check path if it was provided as part of the rule
+                            if let Some(inner_path) = &inner_rule.path {
+                                // If our paths match, check if our contexts match
+                                if path == inner_path && rule.context != inner_rule.context {
+                                    // In this case we are safe in knowing args[0] is what we need.
+                                    let inner_range: Option<Range<usize>> =
+                                        inner_rule.func_call.args[0].0.get_range();
+                                    // error is not None so we have already found something, so we just
+                                    // need to add a new error message
+                                    if let Some(unwrapped_error) = error {
+                                        error = Some(unwrapped_error.add_additional_message(&inner_rule.file,
+                                            inner_range,
+                                            &format!("Found duplicate genfscon rules for filesystem {} with differing contexts: {}", inner_rule.fs_name, inner_rule.context)));
+                                    } else {
+                                        let outer_range: Option<Range<usize>> =
+                                            rule.func_call.args[0].0.get_range();
+                                        // error is none so we need to make a new one
+                                        error = Some(CompileError::new(
+                                            "Duplicate genfscon contexts",
+                                            &rule.file,
+                                            outer_range,
+                                            &format!("Found duplicate genfscon rules for filesystem {} with differing contexts: {}", rule.fs_name, rule.context)));
+                                        // We just made the error we are okay to unwrap it
```

Comment:
Might be cleaner to just do `let tmp_error = CompilerError::new(...);` followed by `error = Some(tmp_error.add_additional_message(...)` in order to avoid creating an option just to immediately discard it?